build.gradle

```diff
@@ -91,9 +91,18 @@ opensearchplugin {
   noticeFile rootProject.file('NOTICE')
 }
 
+configurations {
+  agent
+}
+
+
 dependencies {
   api "com.github.luben:zstd-jni:1.5.6-1"
   api "com.intel.qat:qat-java:1.1.1"
+
+  agent "org.opensearch:opensearch-agent-bootstrap:${opensearch_version}"
+  agent "org.opensearch:opensearch-agent:${opensearch_version}"
+  agent "net.bytebuddy:byte-buddy:1.17.5"
```

Comment:
Can we get this from the version catalog?

Reply:
We surely can, trying to cook the fix as fast as possible

Reply:
Is it worth looking into whether this setup can be done in the core itself and applied across the plugins when running tasks like `internalClusterTest` or `integTest`?

Reply:
> Is it worth looking into whether this setup can be done in the core itself and applied across the plugins when running tasks like internalClusterTest or integTest?

Definitely could be done in core plugin tooling, once the solution proven to work

Reply:
I got the Linux and Mac checks working on https://github.com/opensearch-project/job-scheduler/pull/759. Windows is failing, but I think your PR will fix that: https://github.com/opensearch-project/OpenSearch/pull/17873

Reply:
@cwperks I assume we need the following for windows on JS side?
```
grant {
  permission java.net.NetPermission "accessUnixDomainSocket";
};
```

Reply:
Once https://github.com/opensearch-project/OpenSearch/pull/17882 is merged and a new min distro of windows is available, I will re-run the failing windows checks on https://github.com/opensearch-project/job-scheduler/pull/759 to confirm if that's needed or not.

Reply:
we can try to remove it and give a fresh run.

Reply:
I added a line in the security plugin's plugin-security.policy file in https://github.com/opensearch-project/security/pull/5263, but I'll raise another PR to see if that can be removed from there as well.

Reply:
@prudhvigodithi This line was necessary for the security plugin: https://github.com/opensearch-project/security/pull/5264

Reply:
@cwperks can you please check if security plugin is using `tranport-netty-4` ? From this PR https://github.com/opensearch-project/OpenSearch/pull/17882/files the change was part of `tranport-netty-4`.